website/markdown/posts/2020-04-28-why-tuist/post.mdx

```diff
@@ -0,0 +1,191 @@
+YAML-based project generation is your friend.
+However, if you want to work with simpler concepts and ideas, and be more conventional, Tuist is your tool.
+Our principle is that **simplicity and consistency are vital to scale up projects**.
+
+Some well-known companies that adopted a YAML-approach for the generation of projects ended up building a tool on top of it.
+While that's something large companies can do,
+you might not want to find yourself in the situation where you need to build a tool yourself.
+Moreover,
+having tools on top of other tools adds indirections,
+which in fact,
+complicates optimization and debugging.
+
+## Catch misconfiguration errors early
+
+Xcode delegates catching errors to its build system.
+The problem with that is that approach is that there are certain errors that the build system doesn’t know how to misconfigurations in the project. For example, if there’s a circular dependency, the compilation will fail with an error along the lines of: `“X not found”`. _What would you think as a developer?_ _Is it not found because I didn’t configure the right search paths?_ _Should I change the order in which targets are compiled in my scheme?_ The last thing you’d probably think is that there’s a circular dependency.
+
+To prevent that, Tuist tries to catch errors as early as possible.
+We know developers’ time is a precious asset,
+and they should not spend it debugging Xcode errors unrelated to their changes.
+We can’t catch everything, but we do our best.
+Especially around the dependency graph.
+
+Thanks to it, any developer can add a new framework to the dependency graph with the confidence that if they are doing something wrong, Tuist will catch it. Isn’t that great? Let me repeat that again: **anyone can modify the graph with confidence**. I can’t stress enough how great that is to grow your projects. If you make something wrong, you know that `tuist generate` will tell you.
+
+In the entrepreneurial jargon, we'd say that we are **democratizing scaling up projects**.
+
+## And more yet to come...
+
+The benefit of having knowledge on your project is that we can provide streamlined workflows that leverage Apple's building blocks _(e.g. `xcodebuild`, `simctl`)_.
+Here are some of the features that we are planning,
+and that you'll be able to opt-in easily if you are already using Tuist:
+
+- **Caching:** This is one of the features we are the most excited about building.
+  Unlike build systems like [Buck](https://github.com/facebook/buck) or [Bazel](https://bazel.build/),
+  which are mostly adopted by large companies that can invest in build systems and tooling,
+  we are exploring adding caching at the module level.
+  We'll generate projects where only the targets you plan to work on will be generated ―
+  the rest will be pre-compiled frameworks and libraries that we'll pull from a remote cache.
+  And the beauty of this feature is that you'll be able to opt-in easily if you are already using Tuist.
```

Comment:
It’s important to write this, but sounds a bit repetitive (this is mentioned a few lines above). Maybe change the words?